File: python/src/sdk/models/operations/listpets.py
import dataclasses
from ..shared import error as shared_error
from ..shared import pet as shared_pet
from typing import Optional


@dataclasses.dataclass
class ListPetsQueryParams:
    limit: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'limit', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class ListPetsRequest:
    query_params: ListPetsQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class ListPetsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    error: Optional[shared_error.Error] = dataclasses.field(default=None)
    pets: Optional[list[shared_pet.Pet]] = dataclasses.field(default=None)
    

File: go/pkg/models/operations/listpets.go
package operations

import (
	"github.com/speakeasy-sdks-staging/petstore/pkg/models/shared"
)

type ListPetsQueryParams struct {
	Limit *int `queryParam:"style=form,explode=true,name=limit"`
}

type ListPetsRequest struct {
	QueryParams ListPetsQueryParams
}

type ListPetsResponse struct {
	ContentType string
	Error       *shared.Error
	Pets        []shared.Pet
	StatusCode  int
}
